syntax = "proto2";
package island;

option optimize_for = LITE_RUNTIME;

option java_package = "com.box.island.proto";  
option java_outer_classname = "Constant";  

enum UpgradeType {
    NO_UPGRADE = 0;
    MUST_UPGARDE_APP = 1;
    SHOULD_UPGRADE_APP = 2;
    UPGRADE_RESOURCE = 3;
}

enum GameType {
    G1X1 = 1;
    G1X2 = 2;
    G2X2 = 3;
    G3X0 = 4;
    G4X0 = 5;
}

enum ForceType {
    F_NORMAL = 0;
    F_MISSILE = 1;
}

enum ForceMoveType {
    STAY = 1;
    MOVE = 2;
    RAID = 3;
    RAID_FAILURE = 4;
    STORM = 5;    
}

enum IslandType {
	B_BORN = 0;
    B_NORMAL = 1;
    B_MISSILE = 2;
    B_FORT = 3;
}

enum FightResult {
    WIN = 1;
    LOST = 2;
}


enum PropertyKey {
	P_SPEED = 1;
	P_SUPPLY = 2;
	P_SUPPLY_MAX_TROOPS = 3;
	P_MISSILE_SPEED = 4;
	P_MISSILE_SUPPLY = 5;
	P_FORT_RADIUS = 6;
	P_FORT_ATTACK = 7;
	P_FORT_MAX_TROOPS = 8;
	P_FORT_BORN_TROOPS = 9;
		
}




